x-pack/plugins/security/public/plugin.tsx

```diff
@@ -0,0 +1,148 @@
+import { SecurityNavControlService } from './nav_control';
+import { AccountManagementPage } from './account_management';
+import { AuthenticationService, AuthenticationServiceSetup } from './authentication';
+import { ManagementService, UserAPIClient } from './management';
+
+export interface PluginSetupDependencies {
+  licensing: LicensingPluginSetup;
+  home?: HomePublicPluginSetup;
+  management?: ManagementSetup;
+}
+
+export interface PluginStartDependencies {
+  data: DataPublicPluginStart;
+  management?: ManagementStart;
+}
+
+export class SecurityPlugin
+  implements
+    Plugin<
+      SecurityPluginSetup,
+      SecurityPluginStart,
+      PluginSetupDependencies,
+      PluginStartDependencies
+    > {
+  private sessionTimeout!: ISessionTimeout;
+  private readonly navControlService = new SecurityNavControlService();
+  private readonly securityLicenseService = new SecurityLicenseService();
+  private readonly managementService = new ManagementService();
+  private authc!: AuthenticationServiceSetup;
+
+  public setup(
+    core: CoreSetup<PluginStartDependencies>,
+    { home, licensing, management }: PluginSetupDependencies
+  ) {
+    const { http, notifications, injectedMetadata } = core;
+    const { basePath, anonymousPaths } = http;
+    anonymousPaths.register('/login');
+    anonymousPaths.register('/logout');
+    anonymousPaths.register('/logged_out');
+
+    const tenant = `${injectedMetadata.getInjectedVar('session.tenant', '')}`;
+    const sessionExpired = new SessionExpired(basePath, tenant);
+    http.intercept(new UnauthorizedResponseHttpInterceptor(sessionExpired, anonymousPaths));
+    this.sessionTimeout = new SessionTimeout(notifications, sessionExpired, http, tenant);
+    http.intercept(new SessionTimeoutHttpInterceptor(this.sessionTimeout, anonymousPaths));
+
+    const { license } = this.securityLicenseService.setup({ license$: licensing.license$ });
+
+    this.authc = new AuthenticationService().setup({ http: core.http });
+
+    this.navControlService.setup({
+      securityLicense: license,
+      authc: this.authc,
+    });
+
+    if (management) {
+      this.managementService.setup({
+        license,
+        management,
+        authc: this.authc,
+        fatalErrors: core.fatalErrors,
+        getStartServices: core.getStartServices,
+      });
+    }
+
+    // Home is an optional plugin, and if it's disabled we shouldn't try to fill feature catalog.
+    if (home) {
+      home.featureCatalogue.register({
+        id: 'security',
+        title: i18n.translate('xpack.security.registerFeature.securitySettingsTitle', {
+          defaultMessage: 'Security Settings',
+        }),
+        description: i18n.translate('xpack.security.registerFeature.securitySettingsDescription', {
+          defaultMessage:
+            'Protect your data and easily manage who has access to what with users and roles.',
+        }),
+        icon: 'securityApp',
+        path: '/app/kibana#/management/security',
```

Comment:
This link ultimately navigates to `/app/kibana#/management?section=security`, which just places the user at the root of the management application.

I'm not sure where else we'd want to put them, but we can probably just do away with the `/security` portion of this URL, unless we're going to have our own landing page within management for our security offerings.

I also assume at some point that management won't live within the `kibana` app. I don't think we have a functional test for this catalogue entry, so we'll want to keep track of when this happens so we can update accordingly.

Reply:
> I'm not sure where else we'd want to put them, but we can probably just do away with the /security portion of this URL, unless we're going to have our own landing page within management for our security offerings.

Right, I forgot to mention that we previously could set default app for the section and [made `users` app as default](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/security/public/views/management/users_grid/users.js#L17-L19). It seems we cannot do this anymore with the NP API. I'll switch to `users` URL here for now to keep existing behavior.

@mattkime can you please confirm that it's intentional that new Management API doesn't provide a way to specify default application for the section URL (e.g. previously if user navigated to `#/management/security` we'd forward them to `#/management/security/users` with something like this: `uiRoutes.when(SECURITY_PATH, { redirectTo: USERS_PATH });`)? 

> I also assume at some point that management won't live within the kibana app. I don't think we have a functional test for this catalogue entry, so we'll want to keep track of when this happens so we can update accordingly.

Yeah, I see we have quite a few places in Kibana where `app/kibana#/management/` is hard-coded, so I believe it's safe to assume that whoever will be changing this URL will grep for places like this or setup an automatic forwarder with deprecation warning (and announcement).

Reply:
>can you please confirm that it's intentional that new Management API doesn't provide a way to specify default application for the section URL

Its intentional but I'm happy to have a discussion. You can implement the redirect, I don't see the importance of having it at the api level but I might be unfamiliar with your particular challenge.